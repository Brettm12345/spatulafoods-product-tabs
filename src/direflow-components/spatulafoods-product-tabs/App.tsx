import {Styled} from 'direflow-component';
import React, {FC} from 'react';
import styles from './styles/tailwind.css';
import {DesktopView} from './views/DesktopView';
import {MobileView} from './views/MobileView';
import {products} from './data/products';

export interface AppProps {
  productId: string;
}

const App: FC<AppProps> = ({productId}) => {
  const id = parseInt(productId);
  const product = products?.[id];
  if (!product) return null;
  return (
    <Styled styles={styles}>
      <div className="font-sans">
        <div className="hidden md:block">
          <DesktopView product={product} />
        </div>
        <div className="block md:hidden">
          <MobileView product={product} />
        </div>
      </div>
    </Styled>
  );
};
export default App;
