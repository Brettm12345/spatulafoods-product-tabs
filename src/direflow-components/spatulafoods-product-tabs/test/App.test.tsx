import React from 'react';
import {render, unmountComponentAtNode} from 'react-dom';
import renderer from 'react-test-renderer';
import App, {AppProps} from '../App';

const appProps: AppProps = {
  productId: '6966542565568',
};

it('renders without crashing', () => {
  const div = document.createElement('div');
  render(<App {...appProps} />, div);
  unmountComponentAtNode(div);
});

it('matches snapshot as expected', () => {
  const renderTree = renderer.create(<App {...appProps} />).toJSON();
  expect(renderTree).toMatchSnapshot();
});

it('Renders nothing when theres no product', () => {
  const renderTree = renderer.create(<App productId="1" />).toJSON();
  expect(renderTree).toMatchSnapshot();
});
