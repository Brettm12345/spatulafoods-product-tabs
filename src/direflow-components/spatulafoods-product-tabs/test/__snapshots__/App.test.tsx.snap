// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders nothing when theres no product 1`] = `null`;

exports[`matches snapshot as expected 1`] = `
<div
  className="font-sans subpixel-antialiased"
>
  <div
    className="hidden md:block w-full max-w-4xl mx-auto"
  >
    <div
      aria-orientation="horizontal"
      className="flex mx-auto my-6 p-1 bg-brand-100 rounded-xl space-x-1"
      role="tablist"
    >
      <button
        aria-selected={false}
        className="py-2.5 w-full text-brand-700 text-sm font-bold tracking-wider leading-5 rounded-xl select-none uppercase transition-colors ease-linear focus:outline-none focus:ring-2 ring-white ring-opacity-60 ring-offset-2 ring-offset-brand-400 hover:bg-brand-500/20"
        id="headlessui-tabs-tab-13"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        Cooking Instructions
      </button>
      <button
        aria-selected={false}
        className="py-2.5 w-full text-brand-700 text-sm font-bold tracking-wider leading-5 rounded-xl select-none uppercase transition-colors ease-linear focus:outline-none focus:ring-2 ring-white ring-opacity-60 ring-offset-2 ring-offset-brand-400 hover:bg-brand-500/20"
        id="headlessui-tabs-tab-14"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        Ingredients
      </button>
      <button
        aria-selected={false}
        className="py-2.5 w-full text-brand-700 text-sm font-bold tracking-wider leading-5 rounded-xl select-none uppercase transition-colors ease-linear focus:outline-none focus:ring-2 ring-white ring-opacity-60 ring-offset-2 ring-offset-brand-400 hover:bg-brand-500/20"
        id="headlessui-tabs-tab-15"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        Nutrition Info
      </button>
    </div>
    <div />
  </div>
  <div
    className="block md:hidden"
  >
    <div
      className="mt-4 space-y-3"
    >
      <button
        aria-expanded={false}
        className="text-md no-tap-highlight flex items-center justify-between px-4 py-2 w-full text-left text-brand-900 font-bold bg-brand-100 rounded-lg select-none transition-colors ease-in-out hover:bg-brand-200 focus:outline-none focus:ring focus:ring-brand-300 focus:ring-opacity-75"
        id="headlessui-disclosure-button-19"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        <span>
          Cooking Instructions
        </span>
         
        <svg
          aria-hidden="true"
          className="size-6 text-brand-900 transition-transform duration-200 delay-75 ease-in"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
            fillRule="evenodd"
          />
        </svg>
      </button>
      <button
        aria-expanded={false}
        className="text-md no-tap-highlight flex items-center justify-between px-4 py-2 w-full text-left text-brand-900 font-bold bg-brand-100 rounded-lg select-none transition-colors ease-in-out hover:bg-brand-200 focus:outline-none focus:ring focus:ring-brand-300 focus:ring-opacity-75"
        id="headlessui-disclosure-button-21"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        <span>
          Ingredients
        </span>
         
        <svg
          aria-hidden="true"
          className="size-6 text-brand-900 transition-transform duration-200 delay-75 ease-in"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
            fillRule="evenodd"
          />
        </svg>
      </button>
      <button
        aria-expanded={false}
        className="text-md no-tap-highlight flex items-center justify-between px-4 py-2 w-full text-left text-brand-900 font-bold bg-brand-100 rounded-lg select-none transition-colors ease-in-out hover:bg-brand-200 focus:outline-none focus:ring focus:ring-brand-300 focus:ring-opacity-75"
        id="headlessui-disclosure-button-23"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        <span>
          Nutrition Info
        </span>
         
        <svg
          aria-hidden="true"
          className="size-6 text-brand-900 transition-transform duration-200 delay-75 ease-in"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
            fillRule="evenodd"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
